src/mds/CDir.cc

```diff
@@ -2342,10 +2354,9 @@ void CDir::_committed(int r, version_t v)
     mark_clean();
 
   // dentries clean?
-  for (map_t::iterator it = items.begin();
-       it != items.end(); ) {
-    CDentry *dn = it->second;
-    ++it;
+  for (auto p = dirty_dentries.begin(); !p.end(); ) {
```

Comment:
We should be able to remove `is_dirty` checks below, yes?

Reply:
yes, but only for dn. inode created by MDCache::cow_inode is not dirty